python/tskit/trees.py

```diff
@@ -4244,13 +4135,136 @@ def f2(self, sample_sets, indexes=None, windows=None, mode="site",
             self._ll_tree_sequence.f2, 2, sample_sets, indexes=indexes, windows=windows,
             mode=mode, span_normalise=span_normalise)
 
+    def mean_descendants(self, reference_sets):
+        """
+        Computes for every node the mean number of samples in each of the
+        `reference_sets` that descend from that node, averaged over the
+        portions of the genome for which the node is ancestral to *any* sample.
+        The output is an array, `C[node, j]`, which reports the total span of
+        all genomes in `reference_sets[j]` that inherit from `node`, divided by
+        the total span of the genome on which `node` is an ancestor to any
+        sample in the tree sequence.
+
+        .. note:: This interface *may change*, particularly the normalization by
+            proportion of the genome that `node` is an ancestor to anyone.
+
+        :param iterable reference_sets: A list of lists of node IDs.
+        :return: An array with dimensions (number of nodes in the tree sequence,
+            number of reference sets)
+        """
+        # An array of booleans per tree, true if node u is in the tree else false.
+        in_tree = self.sample_count_stat(
+            [self.samples()], lambda x: x + 1, output_dim=1, polarised=True,
+            strict=False, mode="node", windows="trees")[..., 0] > 1
```

Comment:
I *think* that `ts.segregating_sites(..., mode="node")` does this.

Reply:
Oh, darn: it *almost* does this, but doesn't count root nodes.

Reply:
I played around with segregating_sites for a bit all right, but it was its behaviour around roots that made me give up. I don't see any around that.